core/src/test/java/org/elasticsearch/bwcompat/MixedRepositoryInteroperabilityIT.java

```diff
@@ -0,0 +1,178 @@
+            assertThat(originalSnapshot.snapshotId(), equalTo(new SnapshotId("test_1", SnapshotId.UNASSIGNED_UUID)));
+            assertThat(Sets.newHashSet(originalSnapshot.indices()), equalTo(indices));
+
+            logger.info("--> restore the original snapshot");
+            final Set<String> restoredIndices = Sets.newHashSet(
+                restoreSnapshot(repoName, originalSnapshot.snapshotId().getName())
+            );
+            assertThat(restoredIndices, equalTo(indices));
+
+            final String snapshotName2 = "test_2";
+            logger.info("--> take a new snapshot of the old index");
+            for (int i = 0; i < 10; i++) {
+                index(originalIndex, "doc", Integer.toString(i), "foo", "new-bar-" + i);
+            }
+            refresh();
+            snapshotInfos.add(createSnapshot(repoName, snapshotName2));
+
+            logger.info("--> get the snapshots with the newly created snapshot [{}]", snapshotName2);
+            Set<SnapshotInfo> snapshotInfosFromRepo = Sets.newHashSet(getSnapshots(repoName));
+            assertThat(snapshotInfosFromRepo, equalTo(snapshotInfos));
+            snapshotInfosFromRepo.forEach(snapshotInfo -> {
+                assertThat(Sets.newHashSet(snapshotInfo.indices()), equalTo(indices));
+            });
+
+            final String snapshotName3 = "test_3";
+            final String indexName2 = "index2";
+            logger.info("--> take a new snapshot with a new index");
+            createIndex(indexName2);
+            indices.add(indexName2);
+            for (int i = 0; i < 10; i++) {
+                index(indexName2, "doc", Integer.toString(i), "foo", "new-bar-" + i);
+            }
+            refresh();
+            snapshotInfos.add(createSnapshot(repoName, snapshotName3));
+
+            logger.info("--> get the snapshots with the newly created snapshot [{}]", snapshotName3);
+            snapshotInfosFromRepo = Sets.newHashSet(getSnapshots(repoName));
+            assertThat(snapshotInfosFromRepo, equalTo(snapshotInfos));
+            snapshotInfosFromRepo.forEach(snapshotInfo -> {
+                if (snapshotInfo.snapshotId().getName().equals(snapshotName3)) {
+                    // only the last snapshot has all the indices
+                    assertThat(Sets.newHashSet(snapshotInfo.indices()), equalTo(indices));
+                } else {
+                    assertThat(Sets.newHashSet(snapshotInfo.indices()), equalTo(Sets.newHashSet(originalIndex)));
+                }
+            });
+
+            deleteIndices(indices); // clean up indices before starting again
+        }
+    }
+
+    /**
+     * This tests reading and writing the snapshot index file, where the snapshot ids stored in the index file
+     * could contain a snapshot name + UUID pair(newer versions) or could contain only the snapshot name (old versions).
+     */
+    public void testSnapshotsIndexFile() throws Exception {
+
```

Comment:
Ah yes, good catch :) I added tests for the snapshot index file in https://github.com/elastic/elasticsearch/pull/18228/files/d973e652a47f5a481de49d99744661f98568eea5#diff-32116318faa60fd0d82c4abd0551f352R111

I forgot to remove this method, I will remove it now